### PR TITLE
Data source storage_volumes_files

### LINKED ISSUE #8: [New Data Source]: storage_volumes_files
### Description

This was requested by Email by Jamie Rix from Cisco
This data source should support

volume name
path
byte_offset
overwrite
stream_name

### Requested Resource(s) and/or Data Source(s)

* netapp_ontap_storage_volumes_files_data_source

### Potential Terraform Configuration

_No response_

### References

https://library.netapp.com/ecmdocs/ECMLP2885799/html/#/storage/file_info_collection_get

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by suhasbshekar**
#8

### CODE REVIEW

File: examples/data-sources/netapp-ontap_storage_volumes_filess/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+data "netapp-ontap_storage_volumes_filess_data_source" "storage_volumes_filess" {
```
Comment (wenjun666) [inline examples/data-sources/netapp-ontap_storage_volumes_filess/data-source.tf]: maybe we discussed before, should be "netapp-ontap_storage_volumes_file_data_source" and "netapp-ontap_storage_volumes_files_data_source"?

File: internal/provider/storage_volumes_files_data_source.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+				MarkdownDescription: "svm name",
+				Required:            true,
+			},
+			"path": schema.StringAttribute{
+				MarkdownDescription: "Relative path of a file or directory in the volume",
+				Required:            true,
+			},
+			"byte_offset": schema.Int64Attribute{
+				MarkdownDescription: "The file offset to start reading from",
+				Optional:            true,
```
Comment (wenjun666) [inline internal/provider/storage_volumes_files_data_source.go]: "byte_offset", "overwrite_enabled", "name" can not returned from the API?

Comment (suhasbshekar) [reply internal/provider/storage_volumes_files_data_source.go]: that's right

File: examples/data-sources/netapp-ontap_storage_volumes_filess/data-source.tf (hunk 1 above)
Comment (suhasbshekar) [reply examples/data-sources/netapp-ontap_storage_volumes_filess/data-source.tf]: removed netapp-ontap_storage_volumes_filess_data_source

**Comment by carchi8py** [comment]
This is for 1.2?

File: docs/data-sources/storage_volumes_files_data_source.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  StorageVolumesFiles data source
+---
+
+# netapp-ontap_storage_volumes_files_data_source (Data Source)
+
+Retrieves an existing storage volumes files
+
+## Example Usage
+```terraform
+data "netapp-ontap_storage_volumes_files_data_source" "storage_volumes_files" {
```
Comment (wenjun666) [inline docs/data-sources/storage_volumes_files_data_source.md]: "netapp-ontap_storage_volumes_files"

File: docs/data-sources/storage_volumes_files_data_source.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "netapp-ontap_storage_volumes_files_data_source Data Source - terraform-provider-netapp-ontap"
+subcategory: "Storage"
+description: |-
+  StorageVolumesFiles data source
```
Comment (wenjun666) [inline docs/data-sources/storage_volumes_files_data_source.md]: should be spaces in between StorageVolumesFiles

File: docs/data-sources/storage_volumes_files_data_source.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+---
+
+# netapp-ontap_storage_volumes_files_data_source (Data Source)
```
Comment (wenjun666) [inline docs/data-sources/storage_volumes_files_data_source.md]: netapp-ontap_storage_volumes_files

File: internal/provider/storage/storage_volumes_files_data_source.go [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+			},
+			"byte_offset": schema.Int64Attribute{
+				MarkdownDescription: "The file offset to start reading from",
+				Optional:            true,
+			},
+			"overwrite_enabled": schema.BoolAttribute{
+				MarkdownDescription: "Whether the file can be overwritten",
+				Optional:            true,
+			},
+			"name": schema.StringAttribute{
```
Comment (wenjun666) [inline internal/provider/storage/storage_volumes_files_data_source.go:97]: there are two "name" options, the other one is nested in "storage_volumes_files", do we need the one here?

File: docs/data-sources/storage_volumes_files_data_source.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "netapp-ontap_storage_volumes_files_data_source Data Source - terraform-provider-netapp-ontap"
```
Comment (wenjun666) [inline docs/data-sources/storage_volumes_files_data_source.md]: netapp-ontap_storage_volumes_files

File: internal/provider/storage/storage_volumes_files_data_source.go (hunk 6 above)
Comment (suhasbshekar) [reply internal/provider/storage/storage_volumes_files_data_source.go:97]: yes, one for the input and other for results in a set

**Review by suhasbshekar — COMMENTED** [review]
Made changes on comments

**Review by wenjun666 — APPROVED** [review]

**Pull request merged**